src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/services/reports/ReportsService.java

```diff
@@ -15,22 +16,32 @@
     public static final String TEST_JURISDICTION = "BULKSCAN";
 
     private final EnvelopeCountSummaryRepository repo;
+    private final ZeroRowFiller zeroRowFiller;
 
-    public ReportsService(EnvelopeCountSummaryRepository repo) {
+    // region constructor
+    public ReportsService(
+        EnvelopeCountSummaryRepository repo,
+        ZeroRowFiller zeroRowFiller
+    ) {
         this.repo = repo;
+        this.zeroRowFiller = zeroRowFiller;
     }
+    // endregion
 
     public List<EnvelopeCountSummary> getCountFor(LocalDate date, boolean includeTestJurisdiction) {
-        return repo
-            .getReportFor(date)
+        return zeroRowFiller
+            .fill(repo.getReportFor(date).stream().map(this::fromDb).collect(toList()), date)
```

Comment:
Pushy.. How about reducing unnecessary extra stream?

```java
return zeroRowFiller.fill(
    repo.getReportFor(date)
        .stream()
        .filter(it -> includeTestJurisdiction || !Objects.equals(it.getJurisdiction(), TEST_JURISDICTION))
        .map(this::fromDb)
        .collect(toList()),
    date
);
```

Reply:
filtering need to happen *after* filler does its' thing